blenderproc/python/writer/BopWriterUtility.py

```diff
@@ -82,13 +87,56 @@ def write_bop(output_dir: str, target_objects: Optional[List[MeshObject]] = None
                            f"Either remove the dataset parameter or assign custom property 'bop_dataset_name'"
                            f" to selected objects")
 
+    if calc_mask_info_coco:
+        # It might be that a chunk dir already exists where the writer appends frames.
+        # If one (or multiple) more chunk dirs are created to save the rendered frames to,
+        # mask/info/coco annotations need to be calculated for all of them
+        chunk_dirs = sorted(glob.glob(os.path.join(chunks_dir, '*')))
+        chunk_dirs = [d for d in chunk_dirs if os.path.isdir(d)]
+        starting_chunk_dir = sorted(chunk_dirs)[-1] if chunk_dirs else None
```

Comment:
can we move this part down to l105+ ?

Reply:
ah now I get the logic, you check whether a new chunk_dir was created and only then run the calc_gt_info_mask stuff for the last chunk_dir.

Reply:
hm I don't think so; I'm trying to determine the last chunk dir the writer will be appending to. Let's say you write 25 files per chunk, and in a previous run one only has written 20 frames. gt, info and coco are then only calculated for these 20 frames. if you run the script again and want to render e.g. 40 frames, the writer fills the five remaining frames from the first chunk dir, and creates two additional ones (5 + 25 + 10). before writing we need to determine the currently last chunk dir to be able to also calculate gt info and coco for the 5 additional frames, and all the upcoming chunk dirs.

Reply:
Ah okay, thanks for the explanation. But one chunk is usually 1000 images, are you sure you don't repeat the calculation of the masks/info/coco for the same images? In the BOP Challenge examples the BopWriter is called every 25 frames, i.e. after each scene (!= chunk). 

I don't see where you update the image ids, to me it looks like you compute it for all images in `scene_gt.json` every time, but I could be wrong:

https://github.com/DLR-RM/BlenderProc/blob/5c4e0f1d22aad92ff2e64bf87995c8256bdf9481/blenderproc/python/writer/BopWriterUtility.py#L516-L517